gitwatch.sh

```diff
@@ -120,18 +120,53 @@ is_command () { # Tests for the availability of a command
 if [ -z "$GW_GIT_BIN" ]; then GIT="git"; else GIT="$GW_GIT_BIN"; fi
 if [ -z "$GW_INW_BIN" ]; then INW="inotifywait"; else INW="$GW_INW_BIN"; fi
 
+# if Mac, use fswatch
+if [ "$(uname)" == "Darwin" ]; then
+  INW="fswatch"
+fi
+
 # Check availability of selected binaries and die if not met
 for cmd in "$GIT" "$INW"; do
 	is_command $cmd || { stderr "Error: Required command '$cmd' not found." ; exit 1; }
 done
 unset cmd
 
 # Expand the path to the target to absolute path
-IN=$(readlink -f "$1")
+if [ "$(uname)" == "Darwin" ]; then
+  # implement readlink -f in Darwin
+  # source: https://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac
+  TARGET_FILE=$1
+
+  cd `dirname $TARGET_FILE`
+  TARGET_FILE=`basename $TARGET_FILE`
+
+  # Iterate down a (possible) chain of symlinks
+  while [ -L "$TARGET_FILE" ]
+  do
+    TARGET_FILE=`readlink $TARGET_FILE`
+    cd `dirname $TARGET_FILE`
+    TARGET_FILE=`basename $TARGET_FILE`
+  done
+
+  # Compute the canonicalized name by finding the physical path 
+  # for the directory we're in and appending the target file.
+  PHYS_DIR=`pwd -P`
+  RESULT=$PHYS_DIR/$TARGET_FILE
+  IN=$RESULT
+else
+  IN=$(readlink -f "$1")
+fi
 
```

Comment:
more readable would be if you moved the logic to function, perhaps `follow_link`